langchain4j-vertex-ai-gemini/src/test/java/dev/langchain4j/model/vertexai/VertexAiGeminiStreamingChatModelIT.java

```diff
@@ -367,4 +387,123 @@ void should_accept_function_call() {
         System.out.println("Answer: " + weatherResponse.content().text());
         assertThat(weatherResponse.content().text()).containsIgnoringCase("sunny");
     }
+
+    static class StockInventory {
+        @Tool("Get the stock inventory for a product identified by its ID")
+        public int getStockInventory(@P("ID of the product") String product) {
+            if (product.equals("ABC")) {
+                return 10;
+            } else if (product.equals("XYZ")) {
+                return 42;
+            } else {
+                return 0;
+            }
+        }
+    }
+
+    interface StreamingStockAssistant {
+        @dev.langchain4j.service.SystemMessage(
+            "You MUST call `getStockInventory()` for stock inventory requests.")
+        TokenStream chat(String msg);
+    }
+
+    @Test
+    void should_work_with_parallel_function_calls() throws InterruptedException, ExecutionException, TimeoutException {
+        // given
+        VertexAiGeminiStreamingChatModel model = VertexAiGeminiStreamingChatModel.builder()
+            .project(System.getenv("GCP_PROJECT_ID"))
+            .location(System.getenv("GCP_LOCATION"))
+            .modelName("gemini-1.5-flash-001")
+            .logRequests(true)
+            .logResponses(true)
+            .temperature(0.0f)
+            .topK(1)
+            .build();
+
+        StreamingStockAssistant assistant = AiServices.builder(StreamingStockAssistant.class)
+            .streamingChatLanguageModel(model)
+            .chatMemory(MessageWindowChatMemory.withMaxMessages(10))
+            .tools(new StockInventory())
+            .build();
+
+        // when
+        CompletableFuture<Response<AiMessage>> future = new CompletableFuture<>();
+        assistant.chat("Is there more stock of ABC or of XYZ?")
+            .onNext(System.out::println)
+            .onComplete(future::complete)
+            .onError(future::completeExceptionally)
+            .start();
+        Response<AiMessage> response = future.get(30, TimeUnit.SECONDS);
+
+        // then
+        assertThat(response.content().toString()).contains("XYZ");
+    }
+
+    @Test
+    void should_use_google_search() {
+
+        // given
+        VertexAiGeminiStreamingChatModel modelWithSearch = VertexAiGeminiStreamingChatModel.builder()
+            .project(System.getenv("GCP_PROJECT_ID"))
+            .location(System.getenv("GCP_LOCATION"))
+            .modelName("gemini-1.5-flash-001")
+            .useGoogleSearch(true)
+            .build();
+
+        TestStreamingResponseHandler<AiMessage> handler = new TestStreamingResponseHandler<>();
+
+        // when
+        modelWithSearch.generate("Why is the sky blue?", handler);
+
+        // then
+        assertThat(handler.get().content().text()).containsIgnoringCase("scatter");
```

Comment:
same here

Reply:
Same thing, I haven't found a proper test that doesn't become flaky.